### PR TITLE
implemented ADA feedback for Checkbox

### PR DESCRIPTION
**Pull request by ImmanuelBaskaran**

### CODE REVIEW

**Review by alycrys — APPROVED** [review]

File: packages/lab/src/checkbox/Checkbox.tsx [hunk 1]
```diff
@@ -132,7 +132,8 @@ export const Checkbox = forwardRef<HTMLLabelElement, CheckboxProps>(
         {...rest}
       >
         <input
-          aria-checked={indeterminate ? "mixed" : checked}
+          // aria-checked only needed when indeterminate since native indeterminate behaviour is not used
+          aria-checked={indeterminate ? "mixed" : undefined}
```
Comment (ZarkoPernar) [inline packages/lab/src/checkbox/Checkbox.tsx:136]: Could add a test for this just to prevent breaking this behaviour in some future refactor.

Comment (ImmanuelBaskaran) [reply packages/lab/src/checkbox/Checkbox.tsx:136]: Test has been added

**Review by ZarkoPernar — APPROVED** [review]

File: packages/lab/stories/checkbox/checkbox.stories.tsx [hunk 2]
```diff
@@ -85,6 +46,45 @@ export const Indeterminate: ComponentStory<typeof Checkbox> = () => {
   );
 };
 
+export const Error: ComponentStory<typeof Checkbox> = () => {
+  return (
+    <CheckboxGroup>
+      <Checkbox error checked={false} label="disabled checkbox" />
+      <Checkbox error indeterminate label="disabled indeterminate checkbox" />
+      <Checkbox error checked label="disabled checked checkbox" />
```
Comment (alycrys) [inline packages/lab/stories/checkbox/checkbox.stories.tsx:54]: these are not disabled. label should be corrected

File: packages/lab/stories/checkbox/checkbox.stories.tsx [hunk 3]
```diff
@@ -85,6 +46,45 @@ export const Indeterminate: ComponentStory<typeof Checkbox> = () => {
...
+
+export const Disabled: ComponentStory<typeof Checkbox> = () => {
+  return (
+    <CheckboxGroup>
+      <Checkbox disabled label="disabled checkbox" />
+      <Checkbox
+        disabled
+        indeterminate
+        label="disabled indeterminate checkbox"
+      />
```
Comment (alycrys) [inline packages/lab/stories/checkbox/checkbox.stories.tsx:67]: styling on disabled indeterminate seems to be wrong. shouldn't have hover styling

**Pull request merged**